app/controllers/application_controller/report_downloads.rb

```diff
@@ -169,27 +172,30 @@ def download_csv(view, filename)
     send_data(view.to_csv, :filename => "#{filename}.csv")
   end
 
-  # Send the current report in pdf format
-  def download_pdf(view)
-    render_pdf_internal(view)
+  # Send the current report (displayed in GTL) in pdf format.
+  def download_pdf(report)
+    userid = "#{session[:userid]}|#{request.session_options[:id]}|adhoc"
+    result = report.build_create_results(:userid => userid)
+    render_pdf_internal_rr(report, result)
   end
 
   def report_from_task_id(task_id)
     MiqTask.find(task_id).task_results
   end
 
-  def report_from_report_results(report_result_id)
+  def rrr_from_report_results(report_result_id)
     rr = MiqReportResult.for_user(current_user).find(report_result_id)
-    report = rr.report_results
-    report.report_run_time = rr.last_run_on
-    report
+    report = rr.report_results # TODO: this is strange
+    report.report_run_time = rr.last_run_on # TODO: is this line needed?
```

Comment:
@martinpovolny it appears to be no longer in use, i think we used it to determine name of download file in the past

Reply:
ok, removing it